api/clients/v2/verification/cert_verifier.go

```diff
@@ -130,3 +188,42 @@ func (cv *CertVerifier) GetNonSignerStakesAndSignature(
 
 	return &nonSignerStakesAndSignature, nil
 }
+
+// waitForBlockNumber waits until the internal eth client has advanced to a certain targetBlockNumber.
+//
+// This method will check the current block number of the internal client every CertVerifier.pollInterval duration.
+// It will return nil if the internal client advances to (or past) the targetBlockNumber. It will return an error
+// if the input context times out, or if any error occurs when checking the block number of the internal client.
+func (cv *CertVerifier) waitForBlockNumber(ctx context.Context, targetBlockNumber uint64) error {
+	if cv.pollInterval <= 0 {
+		// don't wait for the internal client to advance
+		return nil
+	}
+
+	ticker := time.NewTicker(cv.pollInterval)
+	defer ticker.Stop()
+
+	var actualBlockNumber uint64
+	for {
+		select {
+		case <-ctx.Done():
+			return fmt.Errorf(
+				"timed out waiting for block number %d (latest block number observed was %d): %w",
+				targetBlockNumber, actualBlockNumber, ctx.Err())
+		case <-ticker.C:
+			actualBlockNumber, err := cv.ethClient.BlockNumber(ctx)
+			if err != nil {
+				return fmt.Errorf("get block number: %w", err)
+			}
+
+			if actualBlockNumber >= targetBlockNumber {
+				return nil
+			}
+
+			cv.logger.Debug(
+				"local client is behind the target block number",
```

Comment:
knit - can we use "reference block" instead?

Reply:
On second thought, if we're making this method public, I don't think "reference block" is appropriate. Someone could use this method in a context which doesn't have anything to do with the reference block in one of our certs.

Thoughts?